sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/DefaultSource.scala

```diff
@@ -172,4 +173,15 @@ class DefaultSource extends FileFormat with DataSourceRegister {
         .mapPartitions(_.map(pair => new String(pair._2.getBytes, 0, pair._2.getLength, charset)))
     }
   }
+
+  private def verifySchema(schema: StructType): Unit = {
+    schema.foreach {
+      field => field.dataType match {
```

Comment:
Oh, I should have said this clearly. Here is a good documentation, (https://github.com/databricks/scala-style-guide#pattern-matching).

I think

```scala
schema.foreach { field =>
  field.dataType match { 
...
```
would be nicer.